File: VB.NET2010/Form1.vb
Public Class Form1

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        Dim iCount As Integer
        Dim i As Integer
        AxSpeechRecognition1.InitControl()

        iCount = AxSpeechRecognition1.GetVoiceCount()
        For i = 0 To iCount - 1
            cboVoice.Items.Add(AxSpeechRecognition1.GetVoiceName(i))

        Next
        If cboVoice.Items.Count > 0 Then
            cboVoice.SelectedIndex = 0
        End If

        iCount = AxSpeechRecognition1.GetOutputWaveFormatCount()
        For i = 0 To iCount - 1
            cbowaveformat.Items.Add(AxSpeechRecognition1.GetOutputWaveFormatName(i))
        Next

        If cbowaveformat.Items.Count > 0 Then
            cbowaveformat.SelectedIndex = 0
        End If
       


    End Sub

    Private Sub btnspeak_Click(sender As System.Object, e As System.EventArgs) Handles btnspeak.Click
        If richTextBox1.Text.Length = 0 Then

            MessageBox.Show("Please enter the text")
            Exit Sub
        End If


        AxSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex)
        AxSpeechRecognition1.Speak(richTextBox1.Text)
    End Sub

    Private Sub btnpause_Click(sender As System.Object, e As System.EventArgs) Handles btnpause.Click
        AxSpeechRecognition1.Pause()
    End Sub

    Private Sub btnresume_Click(sender As System.Object, e As System.EventArgs) Handles btnresume.Click
        AxSpeechRecognition1.Resume()
    End Sub

    Private Sub btnchangeplaybackrate_Click(sender As System.Object, e As System.EventArgs) Handles btnchangeplaybackrate.Click
        AxSpeechRecognition1.Rate = Int32.Parse(txtplaybackrate.Text)
    End Sub

    Private Sub btnvolume_Click(sender As System.Object, e As System.EventArgs) Handles btnvolume.Click
        AxSpeechRecognition1.Volume = Int32.Parse(txtvolume.Text)
    End Sub

    Private Sub btnSavetoWAV_Click(sender As System.Object, e As System.EventArgs) Handles btnSavetoWAV.Click
        If richTextBox1.Text.Length = 0 Then
            MessageBox.Show("Please enter the text")
            Return
        End If

        AxSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex)

        SaveFileDialog1.Filter = "Wave File (*.wav)|*.wav"
        If SaveFileDialog1.ShowDialog() = DialogResult.OK Then

            AxSpeechRecognition1.SetSelectedOutputWaveFormat(cbowaveformat.SelectedIndex)
            AxSpeechRecognition1.SpeakToWave(richTextBox1.Text, SaveFileDialog1.FileName)
            MessageBox.Show("Save Completed")

        End If

    End Sub

    Private Sub btnspeakfromtext_Click(sender As System.Object, e As System.EventArgs) Handles btnspeakfromtext.Click
        OpenFileDialog1.Filter = "Text File (*.txt)|*.txt"
        If OpenFileDialog1.ShowDialog() = DialogResult.OK Then

            Dim sr As System.IO.StreamReader = New System.IO.StreamReader(OpenFileDialog1.FileName)
            richTextBox1.Text = sr.ReadToEnd()
            sr.Close()

            AxSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex)
            AxSpeechRecognition1.SpeakFromText(OpenFileDialog1.FileName)
        End If

    End Sub

    Private Sub btnspeakfromtexttowave_Click(sender As System.Object, e As System.EventArgs) Handles btnspeakfromtexttowave.Click
        OpenFileDialog1.Filter = "Text File (*.txt)|*.txt"
        If OpenFileDialog1.ShowDialog() = DialogResult.OK Then

            Dim sr As System.IO.StreamReader = New System.IO.StreamReader(OpenFileDialog1.FileName)
            richTextBox1.Text = sr.ReadToEnd()
            sr.Close()
        End If


        SaveFileDialog1.Filter = "Wave File (*.wav)|*.wav"
        If SaveFileDialog1.ShowDialog() = DialogResult.OK Then

            AxSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex)
            AxSpeechRecognition1.SetSelectedOutputWaveFormat(cbowaveformat.SelectedIndex)

            AxSpeechRecognition1.SpeakFromTextToWave(OpenFileDialog1.FileName, SaveFileDialog1.FileName)
            MessageBox.Show("Save Completed")
        End If

    End Sub

    Private Sub btnplaywav_Click(sender As System.Object, e As System.EventArgs) Handles btnplaywav.Click
        OpenFileDialog1.Filter = "Wave File (*.wav)|*.wav"
        If OpenFileDialog1.ShowDialog() = DialogResult.OK Then

            AxSpeechRecognition1.SpeakFromWaveFile(OpenFileDialog1.FileName)
        End If

    End Sub

    Private Sub AxSpeechRecognition1_SpeakComplete(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles AxSpeechRecognition1.SpeakComplete
        MessageBox.Show("Speak Completed")
    End Sub

    Private Sub BtnStop_Click(sender As System.Object, e As System.EventArgs) Handles BtnStop.Click
        AxSpeechRecognition1.Stop()
    End Sub
End Class

File: C#2010/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            int i;
            axSpeechRecognition1.InitControl();

            int iCount = axSpeechRecognition1.GetVoiceCount();
            for (i = 0; i < iCount; i++)
                cboVoice.Items.Add(axSpeechRecognition1.GetVoiceName(i));

            if (cboVoice.Items.Count > 0)
                cboVoice.SelectedIndex = 0;

             iCount = axSpeechRecognition1.GetOutputWaveFormatCount();
            for (i = 0; i < iCount; i++)
                cbowaveformat.Items.Add(axSpeechRecognition1.GetOutputWaveFormatName(i));

            if(cbowaveformat.Items.Count >0)
                cbowaveformat.SelectedIndex = 0;



        }

        private void btnspeak_Click(object sender, EventArgs e)
        {
            if (richTextBox1.Text.Length == 0)
            {
                MessageBox.Show("Please enter the text");
                return;
            }

            axSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex);
            axSpeechRecognition1.Speak(richTextBox1.Text);
        }

        private void btnSavetoWAV_Click(object sender, EventArgs e)
        {
            if (richTextBox1.Text.Length == 0)
            {
                MessageBox.Show("Please enter the text");
                return;
            }
            axSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex);

            this.saveFileDialog1.Filter = "Wave File (*.wav)|*.wav";
            if (this.saveFileDialog1.ShowDialog(this) == DialogResult.OK)
            {
                axSpeechRecognition1.SetSelectedOutputWaveFormat(cbowaveformat.SelectedIndex);
                axSpeechRecognition1.SpeakToWave(richTextBox1.Text, saveFileDialog1.FileName);
                MessageBox.Show("Save Completed");

                //axSpeechRecognition1.spe
            }
        }

        private void btnspeakfromtext_Click(object sender, EventArgs e)
        {
            this.openFileDialog1.Filter = "Text File (*.txt)|*.txt";
            if (openFileDialog1.ShowDialog() == DialogResult.OK)
            {
                System.IO.StreamReader sr = new System.IO.StreamReader(openFileDialog1.FileName);
                richTextBox1.Text = sr.ReadToEnd();
                sr.Close();

                axSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex);
                axSpeechRecognition1.SpeakFromText(openFileDialog1.FileName);
            }
        }

        private void btnpause_Click(object sender, EventArgs e)
        {
            axSpeechRecognition1.Pause();
        }

        private void btnresume_Click(object sender, EventArgs e)
        {
            axSpeechRecognition1.Resume();
        }

        private void btnspeakfromtexttowave_Click(object sender, EventArgs e)
        {
              this.openFileDialog1.Filter = "Text File (*.txt)|*.txt";
              if (openFileDialog1.ShowDialog() == DialogResult.OK)
              {
                  System.IO.StreamReader sr = new System.IO.StreamReader(openFileDialog1.FileName);
                  richTextBox1.Text = sr.ReadToEnd();
                  sr.Close();
              }

                this.saveFileDialog1.Filter = "Wave File (*.wav)|*.wav";
            if (this.saveFileDialog1.ShowDialog(this) == DialogResult.OK)
            {
                axSpeechRecognition1.SetSelectedVoice(cboVoice.SelectedIndex);
                axSpeechRecognition1.SpeakFromTextToWave(openFileDialog1.FileName,saveFileDialog1.FileName);
            }
        }

        private void btnplaywav_Click(object sender, EventArgs e)
        {
            this.openFileDialog1.Filter = "Wave File (*.wav)|*.wav";
            if (openFileDialog1.ShowDialog() == DialogResult.OK)
            {
                axSpeechRecognition1.SpeakFromWaveFile(openFileDialog1.FileName);
            }
        }

        private void btnchangeplaybackrate_Click(object sender, EventArgs e)
        {
            axSpeechRecognition1.Rate = Int32.Parse(txtplaybackrate.Text);
        }

        private void btnvolume_Click(object sender, EventArgs e)
        {
            axSpeechRecognition1.Volume = Int32.Parse(txtvolume.Text);
        }

        private void axSpeechRecognition1_SpeakComplete(object sender, EventArgs e)
        {
            MessageBox.Show("Speak Completed");
        }

        private void axSpeechRecognition1_VoiceSentenceStatus(object sender, AxSpeechRecognitionLib._DSpeechRecognitionEvents_VoiceSentenceStatusEvent e)
        {

           
        }

        private void axSpeechRecognition1_VoiceWordsStatus(object sender, AxSpeechRecognitionLib._DSpeechRecognitionEvents_VoiceWordsStatusEvent e)
        {
           
        }

        private void btnstop_Click(object sender, EventArgs e)
        {
            axSpeechRecognition1.Stop();
        }

      
       
    }
}
